conbench/entities/run.py

```diff
@@ -66,8 +67,16 @@ class Run(Base, EntityMixin):
     info: Mapped[Optional[dict]] = Nullable(postgresql.JSONB)
     error_info: Mapped[Optional[dict]] = Nullable(postgresql.JSONB)
     error_type: Mapped[Optional[str]] = Nullable(s.String(250))
-    commit_id: Mapped[str] = NotNull(s.String(50), s.ForeignKey("commit.id"))
-    commit: Mapped[Commit] = relationship("Commit", lazy="joined")
+
+    # The type annotation makes this a nullable many-to-one relationship.
+    # https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#nullable-many-to-one
+    # A nullable many-to-one relationship between Commit (one) and potentially
+    # many Runs, but a run can also _not_ point to a commit.
+    commit_id: Mapped[Optional[str]] = mapped_column(s.ForeignKey("commit.id"))
```

Comment:
FYI: SQLAlchemy infers the 'nullable' from the type annotation :rocket:!

Reply:
For what it's worth, I purposely chose not to do "nullability-by-type-annotations" for other columns: https://github.com/conbench/conbench/pull/859#discussion_r1134517211

But that was more of a spur-of-the-moment decision; I don't really mind if we decide to switch.

Reply:
Ah, nice. I love when dots get connected.

> I think that's a little too magic.

Yeah, I can agree to that. I was impressed this is possible at all :P 

We can use the more explicit style, I like that.